warehouse/views.py

```diff
@@ -134,6 +134,22 @@ def forbidden(exc, request):
 
     # Check if the error has a "result" attribute and if it is a WarehouseDenied
     if hasattr(exc, "result") and isinstance(exc.result, WarehouseDenied):
+        # If the forbidden error is because the user does not have a verified
+        # email address, redirect them to their account page for email verification.
+        if exc.result.reason == "unverified_email":
```

Comment:
I realize you're just following the pattern already established here, but I'm wondering if these should be subclasses of `WarehouseDenied` that we check with `isinstance` instead of comparing the reason here.

Reply:
It's a good thought - I tinkered around with that, but didn't understand exactly what I was doing, since these aren't exceptions, rather subclasses of `pyramid.security.Denied`. I'll leave this as is for now, especially since there's other work in play about restructuring the ACLs and separating authentication from authorization a little better.